src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1117,6 +1117,7 @@ protected override void GetItem(string path)
 
             try
             {
+#if !UNIX
                 bool retrieveStreams = false;
                 FileSystemProviderGetItemDynamicParameters dynamicParameters = null;
```

Comment:
I thought about it more and now have this question: which the following experience is better -- copy `Get-Content -Path <file> -Stream <stream>` to Linux powershell and see the error `"A parameter cannot be found"`, or `"Alternate stream is not supported on this platform"`? I think the latter message is better. To achieve this, maybe we can make the setter of the `Stream` property throw `PlatformNotSupportedException` on non-windows.

As for `#if UNIX/endif` blocks, IMHO, we don't necessarily need to have all codes that only apply to a specific platform guarded by the compile definition. I think it's OK to keep such codes around even for other platforms, and the guideline should be readability and maintainability. I still feel it's better to avoid most of the `#if UNIX` blocks by controlling the entry points:
1. Make sure `-Stream` not available on Unix platforms
2. Throw error when the inline stream syntax is used for `-Path` on Unix platforms
3. Skip the alternate stream discovery codes during remote copy (both `PerformCopyFileFromRemoteSession` and `PerformCopyFileToRemoteSession`) on Unix platforms.

After controlling those entry points (and put some comments for sure), it's safe to know that the alternate stream related code paths will not be hit on Unix platforms.